core/modules/replica/SqlJob.h

```diff
@@ -210,6 +210,34 @@ class SqlJob : public Job {
                                           bool allTables=true,
                                           bool overlapTablesOnly=false) const;
 
+    /**
+     * This version of the table locator method searches for tables where actual
+     * contributions (successful or not) were attempted in a context of the given
+     * transaction. The operation relies upon the persistent records for the tarnsaction
+     * contributions.
+     * 
+     * @param worker The unique identifier of a worker hosting the tables.
+     * @param transactionId The unique identifier of the transaction.
+     * @param table The base name of a table.
+     * @param allTables The optional flag (applies to the partitioned tables only)
+     *   indicating if all or a subset of tables (as determined by the next
+     *   parameter 'overlapTablesOnly') shall be reported. if the flag is set to 'true'
+     *   then the next parameter 'overlapTablesOnly' will be ignored.
+     * @param overlapTablesOnly The optional flag (applies to the partitioned tables only)
+     *   indicating which kind of the partitioned tables to be reported.
+     *   If the flag is set to 'true' then overlap tables will be reported.
+     *   If the flag is set to 'false' then the chunk tables will be reported.
+     *   Note, this parameter is only taken into consideration if the previous
+     *   parameter 'allTables' was set to 'false'.
+     * @return A collection of tables found.
+     * @throw std::invalid_argument If the database or a table isn't valid.
+     */
+    std::vector<std::string> workerTables(std::string const& worker,
+                                          TransactionId const& transactionId,
+                                          std::string const& table,
+                                          bool allTables=true,
+                                          bool overlapTablesOnly=false) const;
```

Comment:
I find the allTables and overlapTablesOnly variables a bit confusing. Maybe get rid of allTables and use:
```
 reportNonOverlapTables = true,
 reportOverlapTables = true
```

Reply:
A "cascading" use of two options `allTables` and `overlapTablesOnly` allows implementing the 3-state logic by unambiguously representing the following table section expression:
```
if (allTables) {
    tables = {"chunk_123", "chunk_123_overlap"};
} else {
    if (overlapTablesOnly) {
        tables = {"chunk_123_overlap"};
    } else {
        tables = {"chunk_123"};
    }
}
```
What you're proposing has 4 states, of which one is undetermined since it would be equivalent to:
```
if (reportNonOverlapTables && reportOverlapTables) {
    tables = {"chunk_123", "chunk_123_overlap"};
} else {
    if (reportOverlapTables && !reportNonOverlapTables) {
        tables = {"chunk_123_overlap"};
    } else if (!reportOverlapTables && reportNonOverlapTables) {
        tables = {"chunk_123"};
   } else {
        // PROBLEM! What are we going to do about this?
        // (reportOverlapTables = false) && (reportNonOverlapTables==false))
   }
}
```